examples/molecular_dynamics/example_molecular_dynamics.cpp

```diff
@@ -0,0 +1,178 @@
+/****************************************************************************
+ * Copyright (c) 2017-2021 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#include <ArborX.hpp>
+
+#include <Kokkos_Random.hpp>
+
+template <class MemorySpace>
+struct Neighbors
+{
+  Kokkos::View<ArborX::Point *, MemorySpace> particles_;
+  float cutoff_radius_;
+};
+
+template <class MemorySpace>
+struct ArborX::AccessTraits<Neighbors<MemorySpace>, ArborX::PredicatesTag>
+{
+  using memory_space = MemorySpace;
+  using size_type = std::size_t;
+  static KOKKOS_FUNCTION size_type size(Neighbors<MemorySpace> const &x)
+  {
+    return x.particles_.extent(0);
+  }
+  static KOKKOS_FUNCTION auto get(Neighbors<MemorySpace> const &x, size_type i)
+  {
+    return attach(intersects(Sphere{x.particles_(i), x.cutoff_radius_}),
+                  (int)i);
+  }
+};
+
+int main(int argc, char *argv[])
+{
+  Kokkos::ScopeGuard guard(argc, argv);
+
+  using ExecutionSpace = Kokkos::DefaultExecutionSpace;
+  using MemorySpace = ExecutionSpace::memory_space;
+
+  ExecutionSpace execution_space{};
+
+  float const dx = 1.f;
+  float const dy = 1.f;
+  float const dz = 1.f;
+  int const nx = 100;
+  int const ny = 100;
+  int const nz = 100;
+  int const n = 4 * nx * ny * nz;
+
+  float const r = 1.f; // cut-off radius
+
+  Kokkos::Profiling::pushRegion("setup");
+  Kokkos::View<ArborX::Point *, MemorySpace> particles(
+      Kokkos::view_alloc("points", Kokkos::WithoutInitializing), n);
+  Kokkos::parallel_for(
+      "make_particles",
```

Comment:
```suggestion
      "Testing::make_particles",
```
or something like that.